concept/thing/thing_manager/validation/validation.rs

```diff
@@ -0,0 +1,75 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use encoding::value::{label::Label, ValueEncodable};
+use lending_iterator::LendingIterator;
+use storage::snapshot::ReadableSnapshot;
+
+use crate::{
+    error::ConceptReadError,
+    thing::{
+        object::ObjectAPI,
+        thing_manager::{validation::DataValidationError, ThingManager},
+    },
+    type_::{
+        annotation::{AnnotationCategory, AnnotationKey, AnnotationUnique},
+        owns::{Owns, OwnsAnnotation},
+        type_manager::TypeManager,
+        Capability, TypeAPI,
+    },
+};
+
+pub(crate) fn get_label_or_concept_read_err<'a>(
```

Comment:
`try_get_label` sounds good, I'll do it!

Reply:
I'll do it for `thing_manager`, but ignore `type_manager` and leave a todo for it, because I'll need to pass `&TypeManager` everywhere, while I'm still using `TypeReader` in most of the places for `type_manager` validations. Haven't found time to refactor it.